README.md

```diff
@@ -1,5 +1,8 @@
+# dd-trace-go
```

Comment:
I noticed that the Go tracer didn't have a title for the repo README.

Reply:
I don't think we need one. That's the repository name. IMHO it looks nicer without this. Happy to discuss it separately from this PR though, if you wish.